### ISSUE TITLE
Conflict between Layout/EmptyLinesAroundAccessModifier and Layout/EmptyLinesAroundClassBody

### ISSUE THREAD

**Issue by natematykiewicz** [rubocop-hq/rubocop#8524]
```yml
Layout/EmptyLinesAroundClassBody:
  EnforcedStyle: empty_lines_special
Layout/EmptyLinesAroundAccessModifier:
  EnforcedStyle: only_before
```

```ruby
class Foo

private
  def bar
  end

end
```

--------

## Expected behavior

This code is valid.

## Actual behavior

This code is impossible to validate because one of the following two offenses is triggered.
> Layout/EmptyLinesAroundClassBody: Empty line missing before first def definition

> Layout/EmptyLinesAroundAccessModifier: Remove a blank line after `private`.

My belief is that the code as-is should be valid and satisfy both rules. But I also believe that `Layout/EmptyLinesAroundClassBody` shouldn't be relevant if the first method is after an access modifier (the new line above `private` satisfies this rule, in my opinion).

RuboCop 0.89.1

Attempting to autocorrect this code sends the autocorrecter into an infinite loop.
